### DISCUSSION TITLE
[Plugin Request] option to generate async function that query and mutate via ApolloClient.

### DISCUSSION THREAD

**Discussion by clord** [dotansimha/graphql-code-generator#4784] [General]
**Is your feature request related to a problem? Please describe.**

Often I have to write boilerplate like this in non-component places of my application: 

```
        const response = await client.mutate<
          RefreshTokenMutation,
          RefreshTokenMutationVariables
        >({
          mutation: RefreshTokenDocument,
        })
```

**Describe the solution you'd like**

similar to `withHooks`, would be nice to have a way to generate a function that bakes these types and the document. Not sure what the best way to pass parameters is, but would look perhaps something like this (could maybe use typescript 4 var-args?) :

```
async function refreshTokenMutation(client: ApolloClient, ... arguments) {
return client.mutate<
          RefreshTokenMutation,
          RefreshTokenMutationVariables
        >({ ... arguments,
          mutation: RefreshTokenDocument,
        })
}
```

**Describe alternatives you've considered**
I currently roll this manually.

**Comment by dotansimha** [comment]
Thanks you @clord ! I'm moving to to Discussions :)
